src/Avalonia.Base/Rendering/Composition/CompositingRenderer.cs

```diff
@@ -253,7 +253,10 @@ private void UpdateCore()
             comp.ClipToBounds = visual.ClipToBounds;
             comp.Clip = visual.Clip?.PlatformImpl;
             comp.OpacityMask = visual.OpacityMask;
-            
+            comp.RenderOptions = comp.Parent != null ? 
```

Comment:
I believe this is a wrong place to merge Render options, since it can be updated later for any ancestor visual. I'd suggest to do the merge in `ServerVisual.Update`